### ISSUE TITLE
Reeling in a laser-grabbed object can wind up grabbing the wrong thing

### ISSUE THREAD

**Issue by chemicalcrux** [Neos-Metaverse/NeosPublic#2499]
## Describe the bug
When you pull a laser-grabbed object close enough, it becomes physically grabbed. However, under some circumstances, this can cause the wrong thing to be grabbed: another Grabbable deeper in the hierarchy.

## Relevant issues
I didn't see anything specifically about this behavior

## To Reproduce
1. Spawn the item from the folder linked below
2. Laser-grab the rock
3. Reel the item in
4. The sphere becomes grabbed

The root of the object has a `Grabbable`, but no collider.

The rock visual has a collider, but no `Grabbable`

The slot containing the sphere has both.

## Expected behavior
The thing being grabbed should not change. I should still be holding onto the root of the object.

## Log Files
[CRUX - 2021.6.14.1343 - 2021-06-17 12_22_40.log](https://github.com/Neos-Metaverse/NeosPublic/files/6671763/CRUX.-.2021.6.14.1343.-.2021-06-17.12_22_40.log)

## Screenshots / Video
https://user-images.githubusercontent.com/8015368/122436966-41f3a600-cf67-11eb-9bc5-ec398b23c5e1.mp4

## Bug information (please complete the following information):
 - How often does it happen: Always
 - Does the bug persist after restarting Neos? Yes
 - Neos Version: 2021.6.14.1343
 - Neos Platform: Windows
 - Link to reproduction World/Item: neosrec:///U-chemicalcrux/R-8162216b-6b6b-467d-ab2a-9ae79fd386c4
 - Regression: Don't think so

## Additional context
The incorrect grab only occurs if there's a collider somewhere on (or probably under) the `Grabbable` component that becomes erroneously grabbed.

The collider can be zero-sized, or even disabled.

The bug doesn't occur if the wrongly-grabbed thing appears later in the list of slots (whether because of a higher OrderOffset or just because it happened to appear later). I'm guessing that it's looking for a collider somewhere in the hierarchy, and doing so in order.

## Reporters:
Just me

**Comment by shiftyscales** [comment]
I feel like you've already answered this one for yourself. I believe the order offset would be the correct way to handle this scenario, because you need to ensure that the collider associated with the object you want to be on the top of the hierarchy is earliest in the hierarchy.

You could also do this by moving a collider onto the top of the hierarchy on the same slot as the first grabbable.

**Comment by H3BO3** [comment]
No, I don't think that should be happening. The same root that was laser-grabbed should end up being physically grabbed.

**Comment by chemicalcrux** [comment]
Yes, I don't see any situation in which it would make sense for you to wind up holding a different `Grabbable`. 

It's not actually checking for a *collision* as far as I can tell -- it's just looking for a collider, full stop. In the video, this causes me to grab something that's way too far away to have collided with my hand.

**Comment by H3BO3** [comment]
I've seen a similar kind of behavior when reeling in a grabbable with a tooltip underneath it - Once it gets to your hand, you equip the tooltip parented inside. That shouldn't happen, imo. I'm unsure if it's directly related to this, and I'll open a second issue if need be

**Comment by shiftyscales** [comment]
> 
> 
> I've seen a similar kind of behavior when reeling in a grabbable with a tooltip underneath it - Once it gets to your hand, you equip the tooltip parented inside. That shouldn't happen, imo. I'm unsure if it's directly related to this, and I'll open a second issue if need be

#766

**Comment by H3BO3** [comment]
While relevant, that still seems like a separate issue with tools not respecting soft permissions for general equip @shiftyscales . The underlying issue of grabbing a tool or grabbable out of a parent grabbable is still another thing imo

**Comment by shiftyscales** [comment]
> While the tool is holstered in the snapper, other users can equip the tool out of the snapper by grabbing the avatar (if he doesn't have it equipped), and pulling the avatar into their hand like they would a tool.

It's literally exactly what you're talking about H3.

> 
> 
> I've seen a similar kind of behavior when reeling in a grabbable with a tooltip underneath it - Once it gets to your hand, you equip the tooltip parented inside. That shouldn't happen, imo. I'm unsure if it's directly related to this, and I'll open a second issue if need be

I was just letting you know that issue already existed for that specific topic you had mentioned.

**Comment by H3BO3** [comment]
@shiftyscales It is not, the title of that issue is "User filter on tool equip". It is requesting a fix for something that otherwise works fine, but is sometimesbroken because of a separate bug - one that is actually the root cause and should be addressed.

**Comment by shiftyscales** [comment]
> 
> 
> @shiftyscales It is not, the title of that issue is "User filter on tool equip". It is requesting a fix for something that otherwise works fine, but is sometimes broken because of a separate bug - one that is actually the root cause and should be addressed.

The issue could be updated/renamed to reflect that, but this issue also sort of covers the same ground. It seems like grabbable events walk the entire hierarchy it seems. That same general issue also has tools in a tooltip multiplexer get equipped out of the multitool.
